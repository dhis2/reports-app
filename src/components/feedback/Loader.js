import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { CircularProgress } from '@dhis2/d2-ui-core'
import { getShowLoading } from '../../redux/selectors/feedback/getShowLoading'

const Loader = props =>
    props.show ? (
        <div className="loading-container">
            <div className="progress-wrap">
                <CircularProgress />
            </div>
            <style jsx>{`
                .loading-container {
                    position: fixed;
                    top: 0;
                    left: 0;
                    height: 100%;
                    width: 100%;
                    background: rgba(0, 0, 0, 0.4);
                    z-index: 3000;
                }
                .progress-wrap {
                    position: absolute;
                    top: 50%;
                    left: 50%;
                    transform: 'translate(-50%, -50%)';
                }
            `}</style>
        </div>
    ) : null

Loader.propTypes = {
    show: PropTypes.bool.isRequired,
}

const mapStateToProps = state => ({
    show: getShowLoading(state),
})

const connectedLoader = connect(mapStateToProps)(Loader)

export { connectedLoader as Loader, Loader as LoaderOriginal }
