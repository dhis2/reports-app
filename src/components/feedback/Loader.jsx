import { CircularProgress } from '@dhis2/d2-ui-core'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { getShowLoading } from '../../redux/selectors/feedback/getShowLoading.js'
import styles from './Loader.module.css'

const Loader = ({ show }) =>
    show ? (
        <div className={styles.loadingContainer}>
            <div className={styles.progressWrap}>
                <CircularProgress />
            </div>
        </div>
    ) : null

Loader.propTypes = {
    show: PropTypes.bool.isRequired,
}

const mapStateToProps = (state) => ({
    show: getShowLoading(state),
})

const connectedLoader = connect(mapStateToProps)(Loader)

export { connectedLoader as Loader, Loader as LoaderOriginal }
