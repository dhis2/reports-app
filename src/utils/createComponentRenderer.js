import React from 'react'
import { shallow } from 'enzyme'

export default function createComponentRenderer(Component, defaultProps) {
    return function(customProps) {
        const props = { ...defaultProps, ...customProps }
        return shallow(<Component {...props} />)
    }
}
