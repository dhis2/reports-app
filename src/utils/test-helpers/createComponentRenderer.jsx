import { shallow } from 'enzyme'
import React from 'react'

export default function createComponentRenderer(Component, defaultProps) {
    return function (customProps) {
        const props = { ...defaultProps, ...customProps }
        return shallow(<Component {...props} />)
    }
}
