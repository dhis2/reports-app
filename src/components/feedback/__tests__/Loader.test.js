import createComponentRenderer from '../../../utils/test-helpers/createComponentRenderer'
import { LoaderOriginal as Loader } from '../Loader'

jest.mock('@dhis2/d2-ui-core', () => ({
    FeedbackSnackbar: 'FeedbackSnackbar',
    CircularProgress: 'CircularProgress',
}))

describe('<Loader />', () => {
    const renderLoader = createComponentRenderer(Loader, {})

    it('should render loader when show is true', () => {
        expect(renderLoader({ show: true })).toMatchSnapshot()
    })

    it('should not render the loader when show is false', () => {
        expect(renderLoader({ show: false })).toMatchSnapshot()
    })
})
