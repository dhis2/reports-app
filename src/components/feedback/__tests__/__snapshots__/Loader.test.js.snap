// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should not render the loader when show is false 1`] = `""`;

exports[`<Loader /> should render loader when show is true 1`] = `
<div
  className="loading-container"
>
  <div
    className="progress-wrap"
  >
    <CircularProgress />
  </div>
  <style>
    
                    .loading-container {
                        position: fixed;
                        top: 0;
                        left: 0;
                        height: 100%;
                        width: 100%;
                        background: rgba(0, 0, 0, 0.4);
                        z-index: 3000;
                    }
                    .progress-wrap {
                        position: absolute;
                        top: 50%;
                        left: 50%;
                        transform: 'translate(-50%, -50%)';
                    }
                
  </style>
</div>
`;
